### PR TITLE
chore: add gh codespaces files

### PR DESCRIPTION
**Pull request by mrhappyma**
Added “devcontainer” files to automatically install extensions and NPM dependencies for codespace users.

I'm not sure if this would actually be used enough to justify being merged, there probably aren't a lot of people working on fig specs in codespaces. Do whatever you want with this PR :)

### CODE REVIEW

**Comment by mrhappyma** [comment]
Codespace commit signing randomly not working again 🫤

File: .devcontainer/base.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+COPY library-scripts/meta.env /usr/local/etc/vscode-dev-containers
+RUN su node -c "umask 0002 && npm install -g ${NODE_MODULES}" \
+    && npm cache clean --force > /dev/null 2>&1
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment if you want to install an additional version of node using nvm
+# ARG EXTRA_NODE_VERSION=10
```
Comment (fedeci) [inline .devcontainer/base.Dockerfile:16]: What is this used for? I suppose it should be removed

**Review by fedeci — COMMENTED** [review]
Apart from the comment I left, I am not against approving this PR.
@SeparateRecords @mschrage, what do you think?

File: .devcontainer/base.Dockerfile (hunk 1 above)
Comment (SeparateRecords) [reply .devcontainer/base.Dockerfile:16]: I'm also not against it! Makes the experience more pleasant for codespace users. There's also precedent, we've previously merged PRs for things like making the DX better for pnpm users.

**Comment by SeparateRecords** [comment]
I'm also not against it! Makes the experience more pleasant for codespace users. There's also precedent, we've previously merged PRs for things like making the DX better for pnpm users.

File: .devcontainer/base.Dockerfile (hunk 1 above)
Comment (mrhappyma) [reply .devcontainer/base.Dockerfile:16]: I'm not quite sure! The devcontainer file generator thing added it, and everything does still build and install and work as before with it removed. I'll go ahead and remove it.

**Comment by mrhappyma** [comment]
👀 @fedeci @SeparateRecords

**Comment by mrhappyma** [comment]
Cool! Thanks 😃 

@SeparateRecords Or @fedeci if you could merge this now 👀

**Review by fedeci — APPROVED** [review]

**Comment by fedeci** [comment]
Thanks!

**Pull request merged**